File: dotnet/tests/amp.tests.integration/Rabbit/TokenConnectionFactoryTests.cs
using System;
using System.Net;
using amp.rabbit;
using amp.rabbit.topology;
using NUnit.Framework;
using RabbitMQ.Client;
using Spring.Context;
using Spring.Context.Support;

namespace amp.tests.integration.Rabbit
{
    [TestFixture]
    public class TokenConnectionFactoryTests
    {
        protected IApplicationContext _context;
        protected TokenConnectionFactory _factory;

        [TestFixtureSetUp]
        public virtual void TestFixtureSetUp()
        {
            _context = new XmlApplicationContext(Config.Authorization.AnubisTwoWaySsl, Config.Topology.GtsSSL);
            _factory = _context.GetObject("IRabbitConnectionFactory") as TokenConnectionFactory;

            //Hack so that we don't have to validate the Anubis server certificate.
            ServicePointManager.ServerCertificateValidationCallback = delegate { return true; };
        }

        [TestFixtureTearDown]
        public virtual void TestFixtureTearDown()
        {
            //Reverse the Hack
            ServicePointManager.ServerCertificateValidationCallback = null;

            _context.Dispose();
        }

        [Test]
        public void Should_be_able_to_get_token_from_Anubis()
        {
            //var rabbitFactory = new Mock<ConnectionFactory>();
            var rabbitFactory = new ConnectionFactory();
            _factory.ConfigureConnectionFactory(rabbitFactory, new Exchange());

            Assert.That(rabbitFactory.UserName, Is.Not.Null);
            Assert.That(rabbitFactory.Password, Is.Not.Null);
        }

        public class TestEvent
        {
            public Guid Id { get; set; }

            public TestEvent()
            {
                Id = Guid.NewGuid();
            }
        }
    }
}

File: java/amp.tests.integration/src/test/java/amp/tests/integration/TokenConnectionFactoryTests.java
package amp.tests.integration;

import java.util.ArrayList;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.springframework.context.support.FileSystemXmlApplicationContext;

import com.rabbitmq.client.ConnectionFactory;

import amp.rabbit.connection.TokenConnectionFactory;
import amp.rabbit.topology.Broker;
import amp.rabbit.topology.Exchange;
import amp.rabbit.topology.ProducingRoute;

import static org.mockito.Mockito.*;

public class TokenConnectionFactoryTests {
    
	protected static FileSystemXmlApplicationContext context;
	protected static TokenConnectionFactory factory;
	
	@BeforeClass
	public static void BeforeAllTests(){
		context = new FileSystemXmlApplicationContext(Config.Authorization.AnubisBasic, Config.Topology.Simple);
		factory = (TokenConnectionFactory) context.getBean("connectionFactory");
	}
	
	@AfterClass
	public static void AfterAllTests(){
		context.close();
	}
	
    @Test
    public void Should_be_able_to_get_token_from_Anubis() throws Exception
    {
    	ConnectionFactory rabbitFactory = mock(ConnectionFactory.class);
		
    	ProducingRoute route = new ProducingRoute(
    			new ArrayList<Broker>(),
    			new Exchange(), 
    			new ArrayList<String>());
    	
    	factory.configureConnectionFactory(rabbitFactory, new Broker());
    	
    	verify(rabbitFactory).setUsername((String)isNotNull());
    	verify(rabbitFactory).setPassword((String)isNotNull());
    }
}
